src/main/java/seedu/address/ui/PersonDetailsPanel.java

```diff
@@ -81,8 +83,10 @@ public void update(Person person) {
             weightDate.setText(WeightCommandMessages.EMPTY_FIELD_WEIGHT_DATE);
             weightValue.setText(WeightCommandMessages.EMPTY_FIELD_WEIGHT_VALUE);
         } else {
-            weightDate.setText(WeightCommandMessages.WEIGHT_DATE_HEADER
-                    + latestWeight.get().getKey().toString());
+            LocalDate numericDate = latestWeight.get().getKey().toLocalDate();
+            DateTimeFormatter numericToText = DateTimeFormatter.ofPattern("MMMM d yyyy");
```

Comment:
The formatter used in the draft PR for the graph is:
```DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd MMM yy");```

Maybe this could be abstracted out into a `seedu.address.commons.util.DateTimeUtil`class?